backend/models.py

```diff
@@ -9,7 +9,7 @@
 
 
 class Course(models.Model):
-    code = models.CharField(max_length=50, primary_key=True)
+    code = models.CharField(max_length=50, primary_key=True, unique=True)
```

Comment:
If you leave this off (and remove the migration), can you double check that it does indeed enforce uniqueness?  There is an SO that suggests that you may still need to add unique=True: http://stackoverflow.com/questions/6039443/primary-key-and-unique-key-in-django